File: apzkr-pzpi-21-11-kokhanevych-pavlo/Task1-Server/src/main/java/com/pet/hotel/data/database/entities/ScheduleEntity.java
package com.pet.hotel.data.database.entities;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.stereotype.Component;

import java.time.LocalDateTime;


@Data
@Entity
@Table(name = "schedule")
@NoArgsConstructor
@Component
public class ScheduleEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "schedule_id")
    @Setter(AccessLevel.PUBLIC)
    private int scheduleId;

    @Column(name = "date_time", nullable = false)
    private LocalDateTime dateTime;

    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name = "diet_id", nullable = false)
    private DietEntity diet;

    public ScheduleEntity(LocalDateTime dateTime, DietEntity diet) {
        this.dateTime = dateTime;
        this.diet = diet;
    }
}

File: apzkr-pzpi-21-11-kokhanevych-pavlo/Task4-MobileClient/src/main/java/com/animal/hotel/data/user/entities/ScheduleEntity.kt
package com.animal.hotel.data.user.entities

import com.animal.hotel.domain.models.user.Schedule
import java.time.LocalDateTime

data class ScheduleEntity(
     val scheduleId: Int,
     val dateTime: LocalDateTime,
     val diet: DietEntity
) {
    fun toSchedule(): Schedule {
        return  Schedule(scheduleId, dateTime, diet.toDiet())
    }

    companion object {
        fun toScheduleEntity(schedule: Schedule) =  ScheduleEntity (
            schedule.scheduleId,
            schedule.dateTime,
            DietEntity.toDietEntity(schedule.diet)
        )
    }
}
